examples/orca_loader/source/orca_loader.cpp

```diff
@@ -110,6 +125,193 @@ class orca_loader_app : public gvk::invokee
+			avk::push_constant_binding_data { avk::shader_type::vertex, 0, sizeof(transformation_matrices) },
+			avk::descriptor_binding(0, 5, mViewProjBuffer),
+			avk::descriptor_binding(4, 4, mImageSamplers),
+			avk::descriptor_binding(7, 9, mMaterialBuffer) 
+		);
+	}
+
+	// Loads an ORCA scene from file or cache file by performing the following steps:
+	//  - Destroy the resources representing the currently loaded scene in n frames
+	//    (where n is the number of frames in flight). The resources to be destroyed are:
+	//     - mDrawCalls
+	//     - mMaterialBuffer
+	//     - mImageSamplers
+	//  - Load ORCA scene from file, creating the resources anew:
+	//     - mDrawCalls
+	//     - mMaterialBuffer
+	//     - mImageSamplers
+	void load_orca_scene_cached(const std::string& aPathToOrcaScene)
+	{
+		// Clean up the current resources, before creating new ones:
+		mOldDrawCalls = std::move(mDrawCalls);
+		mOldImageSamplers =  std::move(mImageSamplers);
+		mOldMaterialBuffer = std::move(mMaterialBuffer);
+		mOldPipeline = std::move(mPipeline);
+		// In #number_of_frames_in_flight() into the future, it will be safe to delete the old resources in render()!
+		// In update() it is not because the fence-wait that ensures that the resources are not used anymore, happens between update() and render().
+		mDestroyOldResourcesInFrame = gvk::context().main_window()->current_frame() + gvk::context().main_window()->number_of_frames_in_flight(); 
+		
+		gvk::orca_scene orca;
+		std::unordered_map<gvk::material_config, std::vector<gvk::model_and_mesh_indices>> distinctMaterialsOrca;
+
+		const std::string cacheFilePath(aPathToOrcaScene + ".cache");
+		auto cacheFileExists = gvk::does_cache_file_exist(cacheFilePath);
+		auto ser = cacheFileExists ?
+			gvk::serializer(gvk::serializer::deserialize(cacheFilePath)) :
+			gvk::serializer(gvk::serializer::serialize(cacheFilePath));
+
+		float start = gvk::fixed_update_timer().absolute_time();
+		float startPart = start;
+		float endPart = 0.0f;
+		std::vector<std::tuple<std::string, float>> times;
+
+		if (!cacheFileExists) {
+			// Load an ORCA scene from file:
+			orca = gvk::orca_scene_t::load_from_file(aPathToOrcaScene);
+			// Get all the different materials from the whole scene:
+			distinctMaterialsOrca = orca->distinct_material_configs_for_all_models();
+			endPart = gvk::fixed_update_timer().absolute_time();
+			times.emplace_back(std::make_tuple("no cache file, load orca file", endPart - startPart));
+			startPart = gvk::fixed_update_timer().absolute_time();
+		}
+
+		// The following loop gathers all the vertex and index data PER MATERIAL and constructs the buffers and materials.
+		// Later, we'll use ONE draw call PER MATERIAL to draw the whole scene.
+		std::vector<gvk::material_config> allMatConfigs;
+		mDrawCalls.clear();
+
+		size_t numIterations = cacheFileExists ? 0 : distinctMaterialsOrca.size();
+		ser.archive(numIterations);
```

Comment:
I think for this, we would have to change the original function, since the convenience overload takes the same parameters. 

Regarding returning the value from `archive`:
Already thought about that during the implementation, but was never shure if it really is more convenient:

`Type` must have default constructor. Currently this is only a problem for BinaryData (used for serialization of buffers/images) which could be solved by template specialization... But if the user ever needs to serialize a type without default constructor, the serializer needs to be changed.

These separate, specialization `archive` versions may be confusing:
```
// This does not modify the argument and either returns the value of distinctMaterialsOrca.size() or the size from the cached file
size_t numIterations = ser.archive(distinctMaterialsOrca.size());

// This modifies the argument. Cannot create buffer in archive and probably does not make sense to return the pointer of the buffer here?
auto mapping = sb->map_memory(avk::mapping_access::write);
ser.archive(ser.binary_data(mapping.get(), texSize));
```
So, if we go this route, maybe it would be better to rename one of the two archive methods. Returning the value does not always lead to less/cleaner code:
```
// Crashes when deserializing, since getModelData is only a placeholder -> needs special attention
auto instances = ser.archive(getModelData().mInstances);

// Possible solution
auto instances = ser.archive(cacheFileExists ? std::vector<gvk::model_instance_data>() : getModelData().mInstances);

// archive does not return the value, not that different from the possible solution
auto instances = cacheFileExists ? std::vector<gvk::model_instance_data>() : getModelData().mInstances;
ser.archive(instances);
```

In the end my thought process was, that having
```
auto someVar;
ser.archive(someVar):
```
everywhere is probably the most consistent way of using the serializer. Of course, I might have missed something and returning the value is not a problem after all.

Reply:
Thinking about the possible different archive versions for variables and buffers:
Instead of letting the user mess with the binary_data class, it would probably be better to just make a serparate function anyways:

```
// Serialize a buffer
template<typename Type>
inline void archive_buffer(Type&& aBuffer, size_t aSize)
{
.....
}

//  Turn this
auto mapping = sb->map_memory(avk::mapping_access::write);
ser.archive(ser.binary_data(mapping.get(), texSize));

// into this
auto mapping = sb->map_memory(avk::mapping_access::write);
ser.archive_buffer(mapping.get(), texSize);
```
Or maybe better names would be: archive_data/archive_memory...

What do you think about that? This would probably solve the confusion between filling a buffer and returning the value. Though I'm still not sure if `size_t numIterations = ser.archive(distinctMaterialsOrca.size());` could be confusing or not.

Reply:
After thinking about this some more and reading your comments, I think it is probably best to _not_ implement the confusing overloads. I think, it is better to leave as it is. The user can write convenience lambdas that serve the same purpose. The serialization-interface shall leave no room for confusion.

Having said that, I think that the code from the `orca_loader` example could be improved quite a bit in terms of readability. In its current state, it might be a bit too densely packed with information or also branching to serve as very understandable example code.

Reply:
Another question regarding the "different archive versions for variables and buffers":

Is there a reason why the serialization-user has to call `map_memory` and then pass the data-pointer and the data-size to `archive`? 

Could an overload(?) to `archive` (or maybe `archive_buffer`) be implemented which calls `map_memory` _internally_? It there's nothing that speaks against it, it could increase convenience and reduce error-proneness.

Reply:
There is not really a reason for letting the user call `map_memory` except, that the serializers current design idea was, to simply take an argument and read or fill it. But I think you are right and it does make sense to have a archive version for `avk::buffer` to make it easier and less error-prone for users.

I think we should probably have something like this:
```
// For general types (i.e int, float, structs, etc)
template<typename Type>
inline void archive(Type&& aValue) {...}

// For buffers with internal mapping
inline void archive_buffer(avk::buffer& aValue) {...}

// For some memory of size (i.e array of vertices, normals, etc)
template<typename Type>
inline void archive_memory(Type&& aValue, size_t aSize) {...}
```
to distinguish between the simple archive function, which only serializes the type it receives, and the more "complicated" archive versions which take memory from somewhere (either through mapping memory of a buffer or pointer). 

Regarding the `orca_loader` code:
Yes, it is now probably more dense that it should be, because it only shows how to use the serializer in the existing code without changing much. I can change the code for this issue, if you want or do you want to make that a separate issue?